practice/src/js/controllers/product-controllers.js

```diff
@@ -1,37 +1,69 @@
-export default class Controller {
+/**
+ * @class ProductController
+ *
+ * Links the user input and the view output.
+ *
+ */
+export default class ProductController {
   constructor(model, view) {
     this.model = model
     this.view = view
 
+    // Explicit this binding
     this.view.renderProduct(this.model.products)
     this.view.bindOpenform()
     this.view.bindCloseform()
     this.view.validator()
-    this.view.bindAddproduct(this.handleAddproduct)
-    this.view.bindDeleteProduct(this.handleDeleteproduct)
-    this.view.bindDeleteAllProduct(this.handleDeleteAllproduct)
-    this.view.bindEditproduct(this.handleEditproduct)
+    this.view.bindAddproduct(this.handleAddProduct)
+    this.view.bindDeleteProduct(this.handleDeleteProduct)
+    this.view.bindDeleteAllProduct(this.handleDeleteAllProduct)
+    this.view.bindEditproduct(this.handleEditProduct)
     this.view.bindSelectedProduct(this.handlerSelectedProduct)
     this.view.bindDeleteSeclectedProduct(this.handlerDeleteSelectedProduct)
   }
 
-  handleAddproduct = (productName, productPrice, productImg, productDes) => {
-    const products = this.model.addProduct(productName, productPrice, productImg, productDes)
-    this.view.renderProduct(products)
+  handleAddProduct = () => {
+    const products = this.view.getValueInput()
+    const productName = products.productName
+    const productPrice = products.productPrice
+    const productImg = products.productImg
+    const productDes = products.productDes
+    if(productName !== '' && productPrice !== '' && productImg !== '' && productDes !== '') {
```

Comment:
Can be written like this: `if(!!productName && !!productPrice && !!productImg && !!productDes)`